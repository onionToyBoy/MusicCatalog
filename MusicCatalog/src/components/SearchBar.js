import React from 'react';
import {View, StyleSheet, TextInput} from 'react-native';

import {colors} from '../constants/colors';

export const SearchBar = ({onSearch}) => {
  return (
    <View style={styles.container}>
      <TextInput
        onChangeText={onSearch}
        style={styles.input}
        placeholder="Введите текст..."
        placeholderTextColor={colors.BRIGHT_GRAY}
      />
    </View>
  );
};
const styles = StyleSheet.create({
  container: {
    //backgroundColor: colors.MIDDLE_GRAY,
    minHeight: 40,
    marginVertical: 10,
  },
  input: {
    borderWidth: 1,
    borderColor: colors.BRIGHT_GRAY,
    flex: 1,
    color: colors.WHITE,
    paddingHorizontal: 15,
  },
});
