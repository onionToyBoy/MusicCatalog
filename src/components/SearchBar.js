import React from 'react';
import { View, StyleSheet, TextInput, Text, TouchableOpacity } from 'react-native';

import { colors } from '../constants/colors';
import { symbols } from '../constants/symbols';

export const SearchBar = ({ onSearch, clearInput, searchValue }) => {
  const iconStyle = searchValue
    ? { color: colors.WHITE, fontSize: 30 }
    : { color: colors.DARK_GRAY };

  return (
    <View style={styles.container}>
      <TextInput
        onChangeText={onSearch}
        style={styles.input}
        value={searchValue}
        placeholder='Enter text...'
        placeholderTextColor={colors.BRIGHT_GRAY}
        maxLength={30}
      />
      <TouchableOpacity data-testid='custom' style={styles.crossContainer} onPress={clearInput}>
        <Text style={iconStyle}>{symbols.CROSS}</Text>
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    minHeight: 40,
    marginVertical: 10,
    borderWidth: 1,
    borderColor: colors.BRIGHT_GRAY,
  },
  input: {
    flex: 1,
    color: colors.WHITE,
    paddingHorizontal: 15,
    fontSize: 17,
  },
  crossContainer: {
    paddingHorizontal: 12,
  },
});
