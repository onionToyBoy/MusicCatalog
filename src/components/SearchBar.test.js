import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

import { SearchBar } from './SearchBar';

describe('SearchBar component', () => {
  let searchValue;
  const onSearch = jest.fn(text => (searchValue = text));
  const clearInput = jest.fn(() => onSearch(''));
  const search = 'deftones';

  test('onSearch should be changed value of TextInput', () => {
    onSearch(search);

    const component = renderer.create(
      <SearchBar onSearch={onSearch} clearInput={clearInput} searchValue={searchValue} />,
    );
    const root = component.root;
    const input = root.findByType('TextInput');

    expect(input.props.value).toBe(search);
  });

  test('clearInput should be removed value of TextInput', () => {
    onSearch(search);
    clearInput();

    const component = renderer.create(
      <SearchBar onSearch={onSearch} clearInput={clearInput} searchValue={searchValue} />,
    );
    const root = component.root;
    const input = root.findByType('TextInput');

    expect(input.props.value).toBe('');
  });
});
