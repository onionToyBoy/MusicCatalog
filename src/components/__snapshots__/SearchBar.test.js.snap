// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "borderColor": "#505050",
      "borderWidth": 1,
      "flexDirection": "row",
      "marginVertical": 10,
      "minHeight": 40,
    }
  }
>
  <TextInput
    allowFontScaling={true}
    maxLength={30}
    placeholder="Enter text..."
    placeholderTextColor="#505050"
    rejectResponderTermination={true}
    style={
      Object {
        "color": "#FFFFFF",
        "flex": 1,
        "fontSize": 17,
        "paddingHorizontal": 15,
      }
    }
    underlineColorAndroid="transparent"
  />
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
        "paddingHorizontal": 12,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#1E1E1E",
        }
      }
    >
      á³µ
    </Text>
  </View>
</View>
`;
